File: python/deleteATBegin.py
# 1.create linked list and delete a node at the beginning of the linked list
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
    
class LinkedList:
    def __init__(self):
        self.head = None
    
    def insert_at_begin(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node
    def delete_at_begin(self):
        temp = self.head
        if self.head is None:
            print("list is empty:")
            return
        self.head = self.head.next
        temp = None
    
    def print_list(self):
        temp = self.head
        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next
        print("None:")


def main():
    ll = LinkedList()
    ll.insert_at_begin(1)
    ll.insert_at_begin(2)
    ll.insert_at_begin(3)
    print("Before deletion:")
    ll.print_list()
    
    ll.delete_at_begin()
    print("After deletion:")
    ll.print_list()

if __name__ == "__main__":
    main()
    

File: javaFile/delete_at_begin.java
class Node {
    int data;
    Node next;
    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;
    LinkedList() {
        this.head = null;
    }

    void insert_node(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node lastNode = head;
        while (lastNode.next != null) {
            lastNode = lastNode.next;
        }
        lastNode.next = newNode;
    }

    void deleteAtBeginning() {
        Node temp = head;
        if (temp == null) {
            System.out.println("List is empty, nothing to delete.");
            return;
        }
        head = head.next;
        temp = null;
    }

    void printList() {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
        System.out.println();
    }

}

public class delete_at_begin {

    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        list.insert_node(98);
        list.insert_node(77);
        list.insert_node(45);
        list.insert_node(23);
        list.insert_node(12);
        System.out.println("Original List:");
        list.printList();
        
        list.deleteAtBeginning();
        System.out.println("After deleting at beginning:");
        list.printList();
    }
}
